libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2245,6 +2245,22 @@ void TypeChecker::typeCheckFunctionGeneralChecks(
 		}
 	}
 
+	if (_functionType->kind() == FunctionType::Kind::BytesConcat)
+	{
+		solAssert(paramArgMap.empty(), "");
+		for (auto const& argument: arguments)
```

Comment:
Should we allow empty arguments? `bytes.concat()`? Is that equivalent to `new bytes(0)`?

Update: apparently `new bytes(0)` is valid.
```
Warning: Statement has no effect.
 --> bytesconcat.sol:3:5:
  |
3 |     new bytes(0);
  |     ^^^^^^^^^^^^
  ```

Reply:
We discussed this on a call, and decided to have no special cases, so we allow empty `concat `, as well as the one with only one argument, but I can change that if we agree we should not allow it.

So, leaving as is?